File: Ejercicios PLAYWRIGHT/PLAYWRIGHTSAUCEDEMO/src/pages/SauceDemoCarritoPage.ts
import { Locator, Page } from '@playwright/test'

export class SauceDemoCarritoPage {
    readonly page:Page;
    readonly btnAddCartProduct1: Locator;
    readonly btnAddCartProduct2: Locator;
    readonly btnCart: Locator;
    readonly btnCheckout: Locator;

    constructor(page:Page) {
        this.page=page;
        this.btnAddCartProduct1 = page.locator('[data-test="add-to-cart-sauce-labs-backpack"]');
        this.btnAddCartProduct2 = page.locator('[data-test="add-to-cart-sauce-labs-bike-light"]');
        this.btnCart = page.locator('xpath=//*[@class="shopping_cart_link"]');
        this.btnCheckout = page.locator('[data-test="checkout"]');
    }

    async Carrito() {
        await this.page.waitForTimeout(3000);
        await this.btnAddCartProduct1.click();
        await this.btnAddCartProduct2.click();
        await this.btnCart.click();
        await this.page.waitForTimeout(3000);
        await this.btnCheckout.click();
        await this.page.waitForTimeout(3000);
    }
}

File: Ejercicios POM/POMSAUCEDEMO/pageobjectmodel/src/main/java/pages/SauceDemoCarritoPage.java
package pages;

import org.openqa.selenium.By;

import actions.Action;
import net.serenitybdd.core.pages.PageObject;
import net.thucydides.core.annotations.DefaultUrl;
import utils.Constants;

@DefaultUrl("https://www.saucedemo.com/inventory.html")

	public class SauceDemoCarritoPage extends PageObject{
		
		By btnAddCartProduct1 = By.name("add-to-cart-sauce-labs-backpack");
		By btnAddCartProduct2 = By.name("add-to-cart-sauce-labs-bike-light");
		By btnCart = By.xpath("//*[@class='shopping_cart_link']");
		By btnCheckout = By.name("checkout");
		
		public void navegateToUrl(String url){
	        Action.navegateToUrl(getDriver(),  Constants.getData(url));
	    }
		
		public void addProductos() {
			
			Action.clickElement(getDriver(), btnAddCartProduct1);
			Action.clickElement(getDriver(), btnAddCartProduct2);
			Action.clickElement(getDriver(), btnCart);
		}
		
		public void comprobarCart() {
			Action.clickElement(getDriver(), btnCheckout);
		}

	}

